brave/src/main/java/brave/Tracing.java

```diff
@@ -295,12 +324,21 @@ public Tracing build() {
       this.stringPropagation = builder.propagationFactory.create(Propagation.KeyFactory.STRING);
       this.currentTraceContext = builder.currentTraceContext;
       this.sampler = builder.sampler;
-      SpanReporter reporter = new SpanReporter(builder.reporter, noop);
+      MutableEndpoint localEndpoint = new MutableEndpoint();
+      localEndpoint.serviceName(builder.localServiceName);
+      localEndpoint.ip(builder.localIp);
+      localEndpoint.port(builder.localPort);
+      Endpoint localZipkinEndpoint = Endpoint.newBuilder()
```

Comment:
note: you'll see stuff like internal use of zipkin2.Endpoint. The first goal is removing that from the user api. Removing from internals will be low-priority refactoring post merge.